internal/auth/oidc/auth_method.go

```diff
@@ -0,0 +1,193 @@
+package oidc
+
+import (
+	"context"
+	"crypto/ed25519"
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/base64"
+	"fmt"
+	"net/url"
+
+	"github.com/hashicorp/boundary/internal/auth/oidc/store"
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/hashicorp/boundary/internal/kms"
+	"github.com/hashicorp/boundary/internal/oplog"
+	wrapping "github.com/hashicorp/go-kms-wrapping"
+	"github.com/hashicorp/go-kms-wrapping/structwrapping"
+	"google.golang.org/protobuf/proto"
+)
+
+// DefaultAuthMethodTableName defines the default table name for an AuthMethod
+const DefaultAuthMethodTableName = "auth_oidc_method"
+
+// AuthMethod contains an OIDC auth method configuration. It is owned
+// by a scope.  AuthMethods can have may assigned Accounts, AudClaims,
+// CallbackUrls, Certificates, SigningAlgs.  AuthMethods also have one State at
+// any given time which determines it's behavior for many its operations.
+type AuthMethod struct {
+	*store.AuthMethod
+	tableName string
+}
+
+// NewAuthMethod creates a new in memory AuthMethod assigned to scopeId.
+// WithMaxAge, WithName and WithDescription are the only valid options. All
+// other options are ignored.
+//
+// State equals the state of the OIDC auth method.  State is not a supported
+// parameter when creating new AuthMethod's since it must be Inactive for all
+// new AuthMethods.
+//
+// DiscoveryUrl equals a URL where the OIDC provider's configuration can be
+// retrieved. The OIDC Discovery Specification requires the URL end in
+// /.well-known/openid-configuration. However, Boundary will strip the URI and
+// only store the scheme and domain.
+//
+// ClientId equals an OAuth 2.0 Client Identifier valid at the Authorization
+// Server.
+//
+// ClientSecret equals the client's secret which will be encrypted when stored
+// in the database and an hmac representation will also be stored when every the
+// secret changes.  The secret is not returned via the API, the hmac is returned
+// so callers can determine if it's been updated.
+//
+// MaxAge equals the Maximum Authentication Age. Specifies the allowable elapsed
+// time in seconds since the last time the End-User was actively authenticated
+// by the OP. If the elapsed time is greater than this value, the OP MUST
+// attempt to actively re-authenticate the End-User. If MaxAge == -1, then it
```

Comment:
"If MaxAge == -1 ..."   should this be <=0  or does "0" also have a special meaning?  My expectation would have been "0" means the user should always be re-authenticated and "-1" means never.

Reply:
Now I see you also documented the option itself.  Consider just referencing the Option's godoc.

Reply:
it's never going to change, so there's no synchronization problem.   max age is documented in the sql, proto, auth method, and option.